browser/search_engines/android/BUILD.gn

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# Import required GN build configuration files
+import("//build/config/android/rules.gni")  # Android build rules
+import("//build/config/locales.gni")
+import("//chrome/common/features.gni")
+import("//tools/grit/grit_rule.gni")
+
+# Define an Android library target for the Quick search engines feature
+android_library("java") {
+  # Java source files to compile
+  sources = [
+    "java/src/org/chromium/brave/browser/SharedPreferencesHelper.java",
+    "java/src/org/chromium/brave/browser/quick_search/ItemTouchHelperCallback.java",
+    "java/src/org/chromium/brave/browser/quick_search/settings/QuickSearchEnginesAdapter.java",
+    "java/src/org/chromium/brave/browser/quick_search/settings/QuickSearchEnginesCallback.java",
+    "java/src/org/chromium/brave/browser/quick_search/settings/QuickSearchEnginesFragment.java",
+    "java/src/org/chromium/brave/browser/quick_search/settings/QuickSearchEnginesModel.java",
+    "java/src/org/chromium/brave/browser/quick_search/settings/QuickSearchEnginesSettingsViewHolder.java",
+    "java/src/org/chromium/brave/browser/quick_search/utils/QuickSearchEnginesUtil.java",
+    "java/src/org/chromium/brave/browser/quick_search/views/QuickSearchEnginesViewAdapter.java",
+    "java/src/org/chromium/brave/browser/quick_search/views/QuickSearchEnginesViewHolder.java",
+  ]
+
+  # Dependencies required by this library
+  deps = [
+    ":java_resources",
+    "//base:supplier_java",
+    "//brave/android/java/org/chromium/chrome/browser/search_engines:java",
+    "//brave/third_party/android_deps:com_google_code_gson_gson_java",
```

Comment:
@bridiver i have created an issue here : https://github.com/brave/brave-browser/issues/45639 to remove the gson usage. i will be working on it as follow up.

Reply:
we would be able to remove it once we are done with playlist changes. it would the only change using gson once i remove gson usage in https://github.com/brave/brave-browser/issues/45639. i was using it as it was included in chromium third_party deps.

Reply:
maybe I'm missing something in my search because the only usage I see in Brave is SharedPreferencesHelper?

Reply:
@bridiver we have playlist changes added as deps here : https://github.com/brave/brave-core/blob/6c063288491774886143b9e2f2138d6f2b505fe4/third_party/android_deps/BUILD.gn#L86. it uses the gson deps.